src/lib/merkle_ledger/util.ml

```diff
@@ -1,38 +1,121 @@
 open Core
 
 module type Inputs_intf = sig
-  module Location : sig
-    type t
-  end
+  module Location : Location_intf.S
 
-  module Account : sig
-    type t
-  end
+  module Key : Intf.Key
+
+  module Account : Intf.Account with type key := Key.t
 
-  module Addr : Merkle_address.S
+  module Hash : Intf.Hash with type account := Account.t
+
+  module Depth : Intf.Depth
 
   module Base : sig
     type t
 
     val get : t -> Location.t -> Account.t option
   end
 
-  val location_of_addr : Addr.t -> Location.t
+  val get_hash : Base.t -> Location.t -> Hash.t
```

Comment:
shouldn't get_hash, set_raw_hash_batch and set_raw_account_batch be inside the Base module? Those functions are main for a Base ledger and should be grouped within the module.